roles/openshift_cloud_provider/tasks/main.yml

```diff
@@ -1,3 +1,17 @@
 ---
+- name: Set cloud provider facts
+  openshift_facts:
+    role: cloudprovider
+    openshift_env: "{{ item | oo_openshift_env }}"
+    openshift_env_structures:
+    - 'openshift.cloudprovider.aws.*'
+    - 'openshift.cloudprovider.openstack.*'
+  with_items:
+  - "{{ hostvars[inventory_hostname] }}"
+  - "{{ hostvars }}"
+
 - include: openstack.yml
-  when: "openshift_cloud_provider is defined and openshift_cloud_provider == 'openstack' and 'provider' in openshift and openshift.provider.name == 'openstack'"
+  when: "'cloudprovider' in openshift and 'kind' in openshift.cloudprovier and openshift.cloudprovider.kind == 'openstack' and 'openstack' in openshift.cloudprovider"
```

Comment:
too bad we can't use the 'fileglob' lookup and loop here...  maybe in ansible 2.x with a block.